client/src/pages/opportunitiespage.cpp

```diff
@@ -82,13 +82,27 @@ void OpportunitiesPage::setupModel()
     treeView()->sortByColumn(nextStepDateColumn, Qt::DescendingOrder);
 }
 
+void OpportunitiesPage::createOpportunity(const QString &accountId)
+{
+    if (!m_createOpportinyFor.isEmpty()) {
+        qWarning() << "There is a pending request on the queue already";
+        return;
+    }
+    m_createOpportinyFor = accountId;
```

Comment:
I don't consider this good design, adding a member variable as a workaround for a missing function call parameter.

Just pass QMap<QString, QString> as argument to createNewItem(), call dataForNewObject() from both callers of createNewItem(), and insert the accountId here before making the call.

As proof that this is better design, it completely removes the possible error case above ("there is a pending request already").

Reply:
done.